src/transformers/models/siglip/image_processing_siglip.py

```diff
@@ -216,6 +216,10 @@ def preprocess(
             size=size,
             resample=resample,
         )
+        # if inputs are all PIL.Image, it is in channel last after conversion to np.ndarray. Save that information to avoid ambiguity.
+        if all(isinstance(image, PIL.Image.Image) for image in images):
+            input_data_format = ChannelDimension.LAST
```

Comment:
Hmmm, I'm not sure about this. Checking the instance for all the inputs for every call is expensive. 

Two comments: 
* We could just check the first image. I don't envision any case when someone is passing in a mix of inputs. Or, if they are, it's not something we should really have to deal with.
* The image processor already provides the `input_data_format` argument which enables the user to override the inference. Instead of this, I suggest just keeping the warning in `infer_channel_dimension_format`

Reply:
I agree. I have removed the check for type and added `input_data_format` to the warning.

Reply:
Although, it will be good if we can get a PIL Image inside the infer_channel_dimension_format so only do the checking if its uncertain. I dont know how much we have to change for this, since now all functions are standardized to use np array

Reply:
> Although, it will be good if we can get a PIL Image inside the infer_channel_dimension_format 

All functions in image_processing_utils accept and return a numpy array (apart from a few exceptions for backwards compatibility). 

Is there a reason you want to add in PIL.Image.Image as an accepted input?

Reply:
> Is there a reason you want to add in PIL.Image.Image as an accepted input?

For inferring the HWC format because torch default is CHW but PIL is HWC. But I understand if it shall not be added for standardization.